perun/collect/trace/locks.py

```diff
@@ -0,0 +1,231 @@
+""" The 'locks' module provides the necessary features for locking some of the resources
```

Comment:
Suggestion: This is well-though functionality and module. I would advise to move it to perun core. Unless it causes some severe issues, but I doubt it.

Reply:
Moved to the `logic` package.